src/components/DeleteComment.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+import { getCommentsByArticleId, deleteComment } from "../api";
+
+ export function DeleteComment({comment, setComments, setClickDelete}) {
+
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [message, setMessage] = useState("");
+
+  const handleDelete = (e) => {
+    e.preventDefault();
+    if(username !== comment.author) {
+        setMessage("Username invalid. Check spelling and ensure you are only trying to delete your own comment!");
+        return;
+    }
+    if(password !== "password") {
+        setMessage("Incorrect password.");
+        return;
+    }
+        deleteComment(comment.comment_id)
+          .then(() => {
+            setComments((currentcomments) => {
+             return currentcomments.filter((item) => item.comment_id !== comment.comment_id )})
+            setMessage("Comment deleted successfully.");
+            setClickDelete(false)
+          })
+          .catch((error) => {
+            console.log(error)
+            setMessage("Error deleting comment:"+ error);
```

Comment:
As a dev it's quite important to be able to see the error itself, but often times websites will choose to show a generic 'something went wrong' or 'error deleting comment' without actually showing the error itself.

It's up to you weather you like this approach or not, but personally I'd say that error messages like this would likely confuse the user more than anything (since we can assume they won't know what axios is)
<img width="390" alt="image" src="https://github.com/123Yusha/FE-Project/assets/152851057/966deee2-d2b8-4c1f-aaf3-4e3afef611c4">